client-docs/README.md

```diff
@@ -0,0 +1,377 @@
+  | {
+      type: "video";
+      variant: "original";
+      name: string;
+      driveId: string;
+    };
+```
+
+### `project.$getProjectSettings`
+
+`() => Promise<ProjectInfo & ProjectSettings>`
+
+Get information about the project and its settings.
+
+### `project.$updateProjectSettings`
+
+`(versionId: string | Array<string>, info: ProjectInfo) => Promise<ProjectInfo & ProjectSettings>`
+
+Update information about the project.
+
+### `project.$leave`
+
+`() => Promise<void>`
+
+Leave the project.
+
+### `project.$sync`
+
+Namespace for getting information about sync and managing sync strategy.
+
+#### `$sync.getState`
+
+`() => Promise<SyncInfo>`
+
+Get information about discovery and sync connection types that are enabled.
+
+#### `$sync.setDiscovery`
+
+`(connectionTypes: ConnectionType[] | null) => Promise<void>`
+
+Set the discovery connection types to enable. If `setDiscovery` has not been previously called with a valid connection type, discovery becomes enabled. If `connectionTypes` is `null` or `[]`, discovery becomes disabled. Do not rely on the resolving of this method to know when the process starts or stops. Instead, listen to the [`'sync-state'`](#sync-state) event on `$sync`.
+
+Note that turning off discovery does not stop existing connections, it only stops the device from searching for and connecting to new devices.
+
+#### `$sync.stop`
+
+_Not implemented_
+
+Force stop discovery and sync.
+
+#### `$sync.setSync`
+
+`(connectionTypes: ConnectionType[] | null) => Promise<void>`
+
+Set the sync connection types to enable. If `setSync` has not been previously called with a valid connection type, sync becomes enabled. If `connectionTypes` is `null` or `[]`, sync becomes disabled. Do not rely on the resolving of this method to know when the process starts or stops. Instead, listen to the [`'sync-state'`](#sync-state) event.
+
+Note that there existing syncing processes with other peers happening, disabling sync does not close these immediately. The server will attempt to gracefully finish or close them and eventually emit the appropriate [`'sync-state'`](#sync-state) event.
+
+#### Events
+
+##### `'sync-state'`
+
+`$sync.addEventListener('sync-state', (info: SyncInfo) => void)`
+
+Emits when the discovery or sync strategy changes.
+
+### `project.$member`
+
+Namespace for managing members of the project. Provides an event emitter-like interface so it can emit and subscribe to events.
+
+#### `$member.invite`
+
+`(deviceId: string, opts: { role: ProjectRole, timeout?: number }) => Promise<InviteResponse>`
+
+Invite a device to the project. Throws if the caller does not have the proper permissions or if the specified `timeout` is reached. Resolves with the response provided by the receiving device.
+
+Accepts the following `opts`:
+
+- `role`: the role to grant for the invited device
+- `timeout`: the maximum amount of time in seconds to wait for a response.
+
+#### `$member.getById`
+
+`(deviceId: string) => Promise<Member>`
```

Comment:
Does returning the `Member` type make sense here, or would it be a more limited subset?

Reply:
I think returning the full type is fine. Limiting a subset is useful when getting over the internet, but since this is all local it's premature optimization until we start benchmarking the nodejs-mobile IPC and seeing where the bottle necks are.